core/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -525,4 +526,20 @@ public void setCompressor(Compressor compressor) {}
       public void setDecompressor(Decompressor decompressor) {}
     }
   }
+
+  /**
+   * Returns a new status with the same code and description, but stripped of any other information
+   * (i.e. cause).
+   *
+   * <p>This is, so that the InProcess transport behaves in the same way as the other transports,
+   * when exchanging statuses between client and server and vice versa.
+   */
+  private static Status stripCause(Status status) {
+    if (status == null) {
+      return null;
+    }
+    return Status
+      .fromCodeValue(status.getCode().value())
```

Comment:
I'm fine with either way. I had wanted to point out that this way has its own advantages.